TFNetworkLayer.py

```diff
@@ -1064,30 +1084,39 @@ def batch_norm(self, data,
         # Note: Our Theano implementation does not use a moving average for this.
         with self.var_creation_scope():
           sample_variance = self.add_param(tf.Variable(
-            initial_value=tf.ones(data.get_bc_spatial_batch_shape()),
+            initial_value=tf.ones(variables_shape),
             name="%s_%s_variance" % (self.name, data.name),
             trainable=False))
         sample_variance = tf.assign_add(sample_variance, (variance - sample_variance) * momentum)
       # If train or if force_sample, use default use_sample=0.0, otherwise use_sample=1.0.
       use_sample = 1.0 + tf.cast(tf.logical_or(self.network.train_flag, force_sample), tf.float32) * (use_sample - 1.0)
       mean = (1. - use_sample) * mean + use_sample * sample_mean
       variance = (1. - use_sample) * variance + use_sample * sample_variance
+      if squeeze_variables:
+        # Double check, that it's not a masked_time case
+        assert not masked_time
```

Comment:
Why is this check needed? This is actually confusing here. This should work no matter if `masked_time` or not `masked_time`. And also don't add redundant comments (which say exactly the same as the code).